### PR TITLE
Initial support for K2 http queries

### PR DESCRIPTION
**Pull request by apolyakov**
Changes it the PR: 

- initialize some common superglobals for all kinds of components
- initialize most of PHP superglobals for K2 server

### CODE REVIEW

File: runtime-light/component/component.cpp [hunk 1]
```diff
@@ -48,8 +51,30 @@ void ComponentState::init_script_execution() noexcept {
 template<ComponentKind kind>
 task_t<void> ComponentState::run_component_prologue() noexcept {
   static_assert(kind != ComponentKind::Invalid);
-
   component_kind_ = kind;
+
+  // common initialization
+  auto &superglobals{php_script_mutable_globals_singleton.get_superglobals()};
```
Comment (astrophysik) [inline runtime-light/component/component.cpp:57]: Maybe move it to separate function like `common_superglobals_initialization`. Because in the future, body of `run_component_prologue` may become too complex.

File: runtime-light/core/globals/php-script-globals.h [hunk 2]
```diff
@@ -8,7 +8,40 @@
...
+inline constexpr auto *REMOTE_ADDR = "REMOTE_ADDR";
+inline constexpr auto *REMOTE_PORT = "REMOTE_PORT";
+
+inline constexpr auto *QUERY_STRING = "QUERY_STRING";
+inline constexpr auto *GATEWAY_INTERFACE = "GATEWAY_INTERFACE";
+
+}; // namespace PhpServerSuperGlobalIndices
+
 struct PhpScriptBuiltInSuperGlobals {
+
```
Comment (astrophysik) [inline runtime-light/core/globals/php-script-globals.h:44]: Unnecessary diff

File: runtime-light/core/globals/php-script-globals.h [hunk 3]
```diff
@@ -8,7 +8,40 @@
 #include "runtime-core/memory-resource/unsynchronized_pool_resource.h"
 #include "runtime-core/runtime-core.h"
 
+namespace PhpServerSuperGlobalIndices {
+
+inline constexpr auto *ARGC = "argc";
```
Comment (astrophysik) [inline runtime-light/core/globals/php-script-globals.h:13]: What is the logic behind the constants location? For example `php-script-globals.h` contains `JOB_ID` that doesn't look generic

File: runtime-light/server/http/init-functions.cpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      }
+      if (std::strstr(header.value.c_str(), ENCODING_DEFLATE) != nullptr) {
+        http_server_ctx.encoding |= HttpServerComponentContext::ENCODING_DEFLATE;
+      }
+    } else if (std::strcmp(header_name.c_str(), HEADER_COOKIE) == 0) {
+      process_cookie_header(header.value, superglobals);
+    } else if (std::strcmp(header_name.c_str(), HEADER_HOST) == 0) {
+      server.set_value(string{SERVER_NAME, std::char_traits<char>::length(SERVER_NAME)}, header.value);
+    } else if (std::strcmp(header_name.c_str(), HEADER_AUTHORIZATION) == 0) {
+      // TODO parse authorization
```
Comment (astrophysik) [inline runtime-light/server/http/init-functions.cpp:101]: Let's add warning in such cases

File: runtime-light/server/http/init-functions.cpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    }
+    // add header entries
```
Comment (astrophysik) [inline runtime-light/server/http/init-functions.cpp:103]: In [`runtime/interface.cpp`](https://github.com/VKCOM/kphp/blob/master/runtime/interface.cpp#L1648) this block stored in else branch

File: runtime-light/server/http/init-functions.cpp [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+    server.set_value(string{REQUEST_URI, std::char_traits<char>::length(REQUEST_URI)}, invoke_http.uri.path);
+  }
+
+  process_headers(std::move(invoke_http.headers), superglobals);
+  // TODO connection_kind, CONTENT_TYPE, is_head_query, _REQUEST, v_POST
+
+  string real_scheme_header{HTTP_X_REAL_SCHEME, std::char_traits<char>::length(HTTP_X_REAL_SCHEME)};
+  string real_host_header{HTTP_X_REAL_HOST, std::char_traits<char>::length(HTTP_X_REAL_HOST)};
+  string real_request_header{HTTP_X_REAL_REQUEST, std::char_traits<char>::length(HTTP_X_REAL_REQUEST)};
+  if (server.isset(real_scheme_header) || server.isset(real_host_header) || server.isset(real_request_header)) {
```
Comment (astrophysik) [inline runtime-light/server/http/init-functions.cpp:175]: In [`runtime/interface.cpp`](https://github.com/VKCOM/kphp/blob/master/runtime/interface.cpp#L1680) condition is different

File: runtime-light/core/globals/php-script-globals.h (hunk 3 above)
Comment (apolyakov) [reply runtime-light/core/globals/php-script-globals.h:13]: I don't actually see it there

File: runtime-light/server/http/init-functions.cpp (hunk 5 above)
Comment (apolyakov) [reply runtime-light/server/http/init-functions.cpp:105]: The semantics is equal in fact

File: runtime-light/server/http/init-functions.cpp (hunk 6 above)
Comment (apolyakov) [reply runtime-light/server/http/init-functions.cpp:179]: Can you please provide a link or an example?

Comment (apolyakov) [reply runtime-light/server/http/init-functions.cpp]: You're right, fixed

**Review by astrophysik — APPROVED** [review]
